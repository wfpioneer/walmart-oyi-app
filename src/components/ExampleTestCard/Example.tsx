import React from 'react';
import {
  Button, Text, TextInput, View
} from 'react-native';

interface ExampleProps{
  userName: string;
}
export function Example(props: ExampleProps) {
  const [name, setUser] = React.useState('');
  const [show, setShow] = React.useState(false);

  return (
    <View>
      <TextInput value={props.userName} onChangeText={setUser} testID="input" />
      <Button
        title="Print Username"
        onPress={() => {
          // let's pretend this is making a server request, so it's async
          // (you'd want to mock this imaginary request in your unit tests)...
          setTimeout(() => {
            setShow(!show);
          }, Math.floor(Math.random() * 200));
        }}
      />
      {show && <Text testID="printed-username">{name}</Text>}
    </View>
  );
}
