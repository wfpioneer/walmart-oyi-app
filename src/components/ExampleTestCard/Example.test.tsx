import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import { Example } from './Example';

test('examples of some things', async () => {
  const {
    getByTestId, getByText, queryByTestId, toJSON
  } = render(<Example userName="Elsa" />);
  const famousProgrammerInHistory = 'Ada Lovelace';

  const input = getByTestId('input');
  fireEvent.changeText(input, famousProgrammerInHistory);

  const button = getByText('Print Username');
  fireEvent.press(button);

  await waitFor(() => expect(queryByTestId('printed-username')).toBeTruthy());

  expect(getByTestId('printed-username').props.children).toBe(
    famousProgrammerInHistory,
  );
  expect(toJSON()).toMatchSnapshot();
});
