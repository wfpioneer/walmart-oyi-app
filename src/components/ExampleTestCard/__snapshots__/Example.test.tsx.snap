// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples of some things 1`] = `
<View>
  <TextInput
    onChangeText={[Function]}
    testID="input"
    value="Elsa"
  />
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Print Username
      </Text>
    </View>
  </View>
  <Text
    testID="printed-username"
  >
    Ada Lovelace
  </Text>
</View>
`;
